lib/developer_portal/app/views/developer_portal/messages/trash/index.html.liquid

```diff
@@ -29,7 +29,7 @@
         <tr>
           <td colspan="4">
             {% if messages.size > 0  %}
-              {{ "Empty the trash" | delete_button: urls.empty_messages_trash, class: 'btn btn-danger' }}
+              {{ "Empty the received trash" | delete_button: urls.empty_messages_trash, class: 'btn btn-danger' }}
```

Comment:
I'd put "Delete received messages"

Because we're not necessarily going to "empty" the trash.

Reply:
Probably, we also need to check the condition on which the button is shown, so that it only appears when there are deleted **received** messages. If all deleted messages are sent ones, we probably don't need to show the button.

Reply:
So, this is due to how sent and received messages are handled... 

Whenever a messages is sent, a `Message` is created, and also one or more `MessageRecipient`s that depend on `Message`.

When we look at the Inbox, we see `MessageRecipient`s, and this way each recipient can manage their received message:
- mark it as read
- move it to trash
- remove it from trash - so it will be gone forever

It's not the same with sent messages. We can move sent message (`Message` instance) to trash, but it we delete it (so, remove from the DB), then all of its dependent `MessageRecipient`s will be gone also.

So, when a developer sends a message to their provider, the provider gets the message in their inbox. But if the developer deletes this sent message forever (which would happen if the trash is cleared completely), then the received message would also get deleted from the provider's inbox :smiling_face_with_tear:  I think this is **not** a desired behavior, so that's why I suggested this.

Alternatively, we could change the data model, so that we can delete `Message` without "killing" `MessageRecipient`s, but I am not sure it's a good idea, as I don't know what would be potential undesired consequences...

Reply:
@mayorova So I would change name to **Delete received messages** for the button.

I am not sure about adding the condition so that it only appears when there are deleted received messages. If all deleted messages are sent ones, we probably don't need to show the button.

may be not showing button on the trash will make it more confusing so better to keep the button and say 
**Delete received messages**  because if any developer will try to open the message still in the trash then will know that they are sent .
so I think we should not hide the button

Reply:
@mayorova thanks for the explanation. It's certainly confusing not being able to remove sent messages. I agree is not good to leave orphaned `MessageRecipient` records in the database. A possible design would be:

- A Message has many `MessageRecipient` and one `MessageSender`
- A `MessageSender` is a new model to show in the user outbox and can be removed like `MessageRecipient` does for inbox
- `Message` is never shown anywhere
- When all `MessageRecipient` and `MessageSender` are removed for a `Message`, then the `Message` is transparently removed as well.

Translated: too much work.

Reply:
@jlledom Yes, totally, that would be a proper solution, but I agree - too much work for a questionable benefit... 
Well, actually there **would** be a benefit as it would allow deleting (forever) both received and sent messages.
However, currently it has not been requested (I haven't found any JIRA for that). In fact, in the admin portal we do not provide an option to delete any of messages from Trash.

We do have an RFE to delete messages via API: [THREESCALE-7346](https://issues.redhat.com/browse/THREESCALE-7346), but it does not specify whether this is to move messages to trash, or to delete them completely.

So, I agree, this would be the best way to solve this, but at this point it's not worth putting so much effort into this, IMO.